syncano_cli/sync/commands.py

```diff
@@ -1,92 +1,109 @@
-    def run(cls, context):
-        """
-        Pull configuration from syncano and store it in current directory.
-        Updates syncano.yml configuration file, and places scripts in scripts
-        directory.
-        When syncano.yml file exists. It will pull only objects defined in
-        configuration file. If you want to pull all objects from syncano use
-        -a/--all flag.
-        """
-        con = syncano.connect(api_key=context.key,
-                              instance_name=context.instance)
-        context.project.update_from_instance(con, context.all,
-                                             context.classes, context.scripts)
-        context.project.write(context.file)
-
-
-class Watch(six.with_metaclass(CommandContainer)):
-    namespace = SYNC_NAMESPACE
-
-    @classmethod
-    @argument('instance', help="Source instance name", action=EnvDefault,
-              envvar='SYNCANO_INSTANCE')
-    def run(cls, context):
-        """
-        Push configuration to syncano. After that  watch for changes in
-        syncano.yml file and scripts and push changed items to syncano.
-        """
-        context.classes = None
-        context.scripts = None
-        context.all = True
-        context.project.timestamp = 0
-        Push.run(context)
-        LOG.info("Watching for file changes")
-        observer = Observer()
-        observer.schedule(ProjectEventHandler(context), path='.', recursive=True)
-        observer.start()
-        try:
-            while True:
-                time.sleep(1)
-        except KeyboardInterrupt:
-            observer.stop()
-        observer.join()
+
+@click.group()
+@click.pass_context
+def top_sync(context):
+    pass
+
+
+@top_sync.group()
+@click.pass_context
+@click.option('-f', '--file', default='syncano.yml', help=u'Instance configuration file.')
+@click.option('--config', help=u'Account configuration file.')
+@click.option('--key', '--key', default=os.environ.get('SYNCANO_API_KEY', None),
+              help=u'override ACCOUNT_KEY used for authentication.')
+def sync(context, file, config, key):
+    """
+    Command for syncing data - classes and scripts
+    :param context:
+    :param file: file which will be used for syncing
+    :param config: the config path - the cli config will be stored there
+    :param key: the Syncano API key
+    :return:
+    """
+    from syncano_cli.main import ACCOUNT_CONFIG, ACCOUNT_CONFIG_PATH
+    config = config or ACCOUNT_CONFIG_PATH
+    ACCOUNT_CONFIG.read(config)
+    context.obj['file'] = file
+    context.obj['config'] = config
+    context.obj['key'] = key or ACCOUNT_CONFIG.get('DEFAULT', 'key')
+    context.obj['project'] = Project.from_config(context.obj['file'])
+
+
+@sync.command()
+@click.pass_context
+@click.option('-s', '--script', help=u"Pull only this script from syncano", multiple=True)
+@click.option('-c', '--klass', help=u"Pull only this class from syncano", multiple=True)
```

Comment:
This should be a `--class`. We use `class` everywhere in syncano. And we only use `klass` internally as it is a reserved keyword.